core/utils/type-utils/src/main/java/datawave/data/type/BaseType.java

```diff
@@ -9,16 +9,25 @@
 
 public class BaseType<T extends Comparable<T> & Serializable> implements Serializable, Type<T>, ObjectSizeOf {
     
-    private static final long serialVersionUID = 5354270429891763693L;
+    private static final long serialVersionUID = 1261391800060720465L;
     private static final long STATIC_SIZE = PrecomputedSizes.STRING_STATIC_REF + Sizer.REFERENCE + Sizer.REFERENCE;
     
     protected T delegate;
     protected String normalizedValue;
     protected final Normalizer<T> normalizer;
     
-    public BaseType(String delegateString, Normalizer<T> normalizer) {
+    /**
+     * Default constructor. Generates the normalized and non-normalized variants of the provided value
+     * 
+     * @param value
+     *            the value
+     * @param normalizer
+     *            the normalizer
+     */
+    public BaseType(String value, Normalizer<T> normalizer) {
         this.normalizer = normalizer;
-        setDelegate(normalizer.denormalize(delegateString));
+        this.delegate = normalizer.denormalize(value);
+        this.normalizedValue = normalizer.normalizeDelegateType(delegate);
```

Comment:
Ok, maybe this makes sense to someone that works in this module a lot, but to me I see "set the delegate to the denormalized, then nomalizedValue= normalize the denormalized". Maybe I'm reading that wrong, or maybe our naming just isn't great.

Reply:
The input string may or may not be normalized.
The input string is then denormalized into a delegate (non-normalized)
The the delegate is then run through the normalizer to produce a normalized value. 

I believe you read that correctly, and you are also correct that the naming is not great.